cdap-app-fabric/src/main/java/co/cask/cdap/security/TokenSecureStoreUpdater.java

```diff
@@ -60,16 +61,23 @@
  */
 public final class TokenSecureStoreUpdater implements SecureStoreUpdater {
   private static final Logger LOG = LoggerFactory.getLogger(TokenSecureStoreUpdater.class);
+
   private final YarnConfiguration hConf;
   private final LocationFactory locationFactory;
+  private final co.cask.cdap.api.security.store.SecureStore secureStore;
   private final long updateInterval;
   private final boolean secureExplore;
+  private final boolean supportsKMS;
 
   @Inject
-  public TokenSecureStoreUpdater(YarnConfiguration hConf, CConfiguration cConf, LocationFactory locationFactory) {
+  TokenSecureStoreUpdater(YarnConfiguration hConf, CConfiguration cConf,
+                                 LocationFactory locationFactory,
+                                 co.cask.cdap.api.security.store.SecureStore secureStore) {
     this.hConf = hConf;
     this.locationFactory = locationFactory;
+    this.secureStore = secureStore;
     secureExplore = cConf.getBoolean(Constants.Explore.EXPLORE_ENABLED) && UserGroupInformation.isSecurityEnabled();
+    supportsKMS = AbstractSecureStore.isKMSBacked(cConf) && AbstractSecureStore.isKMSCapable();
```

Comment:
They were in a Util class, Bhooshan suggested I move them to an abstract class :)

> On Jul 19, 2016, at 10:42 AM, Terence Yim <notifications@github.com> wrote:
> 
> In cdap-app-fabric/src/main/java/co/cask/cdap/security/TokenSecureStoreUpdater.java <https://github.com/caskdata/cdap/pull/6066#discussion_r71386857>:
> 
> >      secureExplore = cConf.getBoolean(Constants.Explore.EXPLORE_ENABLED) && UserGroupInformation.isSecurityEnabled();
> > +    supportsKMS = AbstractSecureStore.isKMSBacked(cConf) && AbstractSecureStore.isKMSCapable();
> Move those static helper methods to SecureStoreUtils as I suggested above.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/caskdata/cdap/pull/6066/files/e867ab05183313ef992bdfd396e440874e647424#r71386857>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAFGk6YT-7JYiDP0Js_d3YFHwwreIVj0ks5qXQxugaJpZM4JOEs0>.
>